snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/ecommerce/events/CheckoutStep.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2015-2023 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.ecommerce.events
+
+import com.snowplowanalytics.core.constants.Parameters
+import com.snowplowanalytics.core.constants.TrackerConstants
+import com.snowplowanalytics.core.ecommerce.EcommerceAction
+import com.snowplowanalytics.snowplow.event.AbstractSelfDescribing
+import com.snowplowanalytics.snowplow.payload.SelfDescribingJson
+
+/**
+ * Track a checkout step.
+ *
+ * @param step Checkout step index.
+ * @param shippingPostcode Shipping address postcode.
+ * @param billingPostcode Billing address postcode.
+ * @param shippingFullAddress Full shipping address.
+ * @param billingFullAddress Full billing address.
+ * @param deliveryProvider Can be used to discern delivery providers e.g. DHL, PostNL etc.
+ * @param deliveryMethod Store pickup, standard delivery, express delivery, international, etc.
+ * @param couponCode Coupon applied at checkout.
+ * @param accountType Type of account used on checkout, e.g. existing user, guest.
+ * @param paymentMethod Any kind of payment method the user selected to proceed. Card, PayPal, Alipay etc.
+ * @param proofOfPayment E.g. invoice or receipt.
+ * @param marketingOptIn If opted in to marketing campaigns to the email address.
+ */
+class CheckoutStep @JvmOverloads constructor(
```

Comment:
Nit: The name `CheckoutStep` does not immediately make it clear whether this is an event or an entity. In the mobile media APIs, I decided to add the `Event` suffix to all event classes and `Entity` to the end of all entity classes. So my suggestion would be to name this `CheckoutStepEvent` to make it clear that it can be tracked. We could also rename `ProductEntity` but up to your consideration (I used the entity suffix in the media APIs mostly to avoid clashes with some Android API names such as `MediaPlayer`).